### PR TITLE
Messages rework

### PR DESCRIPTION
**Pull request by Keruspe**
Make the distinction between BasicGetMessage and ConsumerMessage, reexport those in lapin-futures

### CODE REVIEW

File: async/src/message.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+pub trait ExpandableMessage {
+  fn data(&mut self) -> &mut Vec<u8>;
+
+  fn receive_content(&mut self, data: Vec<u8>) {
+    self.data().extend(data);
+  }
+}
+
+#[derive(Clone,Debug,PartialEq)]
+pub struct ConsumerMessage {
```
Comment (michaelklishin) [inline async/src/message.rs]: Many other clients use `Delivery` as the name.

**Comment by michaelklishin** [comment]
How are they meaningfully different? `basic.get` response can be empty but when there is a message, it's no different from what's delivered to consumers.

**Comment by Keruspe** [comment]
Basic get response carries the number of messages left in the queue

**Comment by michaelklishin** [comment]
Yes but the message is no different. I suggest that you add a structure (in fact, two) for `basic.get` responses which has the message as one of the fields instead. Alternatively it is possible to return a pair/triple from `basic_get`, [Bunny does that](http://rubybunny.info/articles/queues.html).

This client already does a couple of opinionated things but this is a very important part of the public API.

**Comment by Keruspe** [comment]
I was thinking of a tuple too, indeed. I'll probably do that

**Comment by Keruspe** [comment]
I'm open to bikeshedding regarding the name of `BasicGetMessage`

**Pull request merged**